pgml-extension/src/bindings/transformers/whitelist.rs

```diff
@@ -0,0 +1,233 @@
+use std::{error::Error, fmt};
+
+#[cfg(any(test, feature = "pg_test"))]
+use pgrx::{pg_schema, pg_test};
+use serde_json::Value;
+
+use crate::config::get_config;
+
+static CONFIG_HF_WHITELIST: &str = "pgml.huggingface_whitelist";
+static CONFIG_HF_TRUST_REMOTE_CODE_BOOL: &str = "pgml.huggingface_trust_remote_code";
+static CONFIG_HF_TRUST_WHITELIST: &str = "pgml.huggingface_trust_remote_code_whitelist";
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub enum WhitelistError {
+    NotInWhitelist,
+    RemoteCodeNotTrusted,
+}
+
+impl fmt::Display for WhitelistError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            WhitelistError::NotInWhitelist => writeln!(f, "model not in whitelist"),
+            WhitelistError::RemoteCodeNotTrusted => writeln!(f, "model remote code not trusted"),
+        }
+    }
+}
+
+impl Error for WhitelistError {}
+
+/// Verify that the model in the task JSON is allowed based on the huggingface whitelists.
+pub fn verify_task_against_whitelist(task: &Value) -> Result<(), WhitelistError> {
+    let task_model = match get_model_name(task) {
+        Some(model) => model.to_string(),
+        None => return Ok(()),
+    };
+    let whitelisted_models = config_csv_list(CONFIG_HF_WHITELIST);
+
+    let model_is_allowed =
+        whitelisted_models.is_empty() || whitelisted_models.contains(&task_model);
+    if !model_is_allowed {
+        return Err(WhitelistError::NotInWhitelist);
+    }
+
+    let task_trust = get_trust_remote_code(task);
+    let trust_remote_code = get_config(CONFIG_HF_TRUST_REMOTE_CODE_BOOL)
+        .map(|v| v == "true")
+        .unwrap_or(false);
+
+    let trusted_models = config_csv_list(CONFIG_HF_TRUST_WHITELIST);
+
+    let model_is_trusted = trusted_models.is_empty() || trusted_models.contains(&task_model);
+
+    let remote_code_allowed = trust_remote_code && model_is_trusted;
+    if !remote_code_allowed && task_trust == Some(true) {
+        return Err(WhitelistError::RemoteCodeNotTrusted);
+    }
+
+    Ok(())
+}
+
+fn config_csv_list(name: &str) -> Vec<String> {
+    match get_config(name) {
+        Some(value) => value
+            .trim_matches('"')
+            .split(',')
+            .filter_map(|s| {
+                if s.is_empty() {
+                    None
+                } else {
+                    Some(s.to_string())
+                }
+            })
+            .collect(),
+        None => vec![],
+    }
+}
+
+fn get_model_name(task: &Value) -> Option<&str> {
```

Comment:
These JSON accessors make sense to me. I would love to one day have a full Task struct so we can parse once, and then just work with in Rust, but I'm not sure that'll make sense as long as the Python API is in play...